src/WebRTC/SessionDescriptionHandler.js

```diff
@@ -0,0 +1,487 @@
+
+  /**
+   * Destructor
+   */
+  close: {writable: true, value: function () {
+    this.logger.log('closing PeerConnection');
+    // have to check signalingState since this.close() gets called multiple times
+    if(this.peerConnection && this.peerConnection.signalingState !== 'closed') {
+      // TODO: getLocalStreams && getRemoteStreams are deprecated
+      this.peerConnection.getLocalStreams().forEach(function(stream) {
+        stream.getTracks().forEach(function(track) {
+          track.stop();
+        });
+      });
+      this.peerConnection.getRemoteStreams().forEach(function(stream) {
+        stream.getTracks().forEach(function(track) {
+          track.stop();
+        });
+      });
+      this.peerConnection.close();
+    }
+  }},
+
+  /**
+   * Gets the local description from the underlying media implementation
+   * @param {Object} [options] Options object to be used by getDescription
+   * @param {MediaStreamConstraints} [options.constraints] MediaStreamConstraints https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamConstraints
+   * @param {Object} [options.peerConnectionOptions] If this is set it will recreate the peer connection with the new options
+   * @param {Array} [modifiers] Array with one time use description modifiers
+   * @returns {Promise} Promise that resolves with the local description to be used for the session
+   */
+  getDescription: {writable: true, value: function (options, modifiers) {
+    var self = this;
+
+    if (this.session.disableRenegotiation) {
+      this.logger.warn("The flag \"disableRenegotiation\" is set to true for this session description handler. We will not try to renegotiate.");
+      return SIP.Utils.Promise.reject(new SIP.Exceptions.RenegotiationError("disableRenegotiation flag set to true for this session description handler"));
+    }
+
+    options = options || {};
+    if (options.peerConnectionOptions) {
+      this.initPeerConnection(options.peerConnectionOptions);
+    }
+
+    // Merge passed constraints with saved constraints and save
+    this.constraints = Object.assign(this.constraints, options.constraints);
+    this.constraints = this.checkAndDefaultConstraints(this.constraints);
+
+    modifiers = modifiers || [];
+    if (!Array.isArray(modifiers)) {
+      modifiers = [modifiers];
+    }
+    modifiers = modifiers.concat(this.modifiers);
+
+    // Check to see if the peerConnection already has a local description
+    if (this.peerConnection.localDescription && this.peerConnection.localDescription.sdp && this.peerConnection.localDescription.sdp !== '') {
+      return this.createOfferOrAnswer({}, modifiers).then(function(sdp) {
+        return {
+          body: sdp,
+          contentType: self.CONTENT_TYPE
+        };
+      });
+    }
+
+    // GUM and set myself up
+    self.logger.log('acquiring local media');
+    // TODO: Constraints should be named MediaStreamConstraints
+    return this.acquire(self.constraints)
```

Comment:
what about providing already acquired mediastream?

Reply:
You will need to write your own sessionDescriptionHandler to handle that. It is not something we want to deal with on the default implementation.